stubs/Authlib/authlib/jose/rfc7518/jwe_algs.pyi

```diff
@@ -0,0 +1,65 @@
+from _typeshed import Incomplete
+
+from authlib.jose.rfc7516 import JWEAlgorithm
+from authlib.jose.rfc7518 import ECKey
+
+class DirectAlgorithm(JWEAlgorithm):
+    name: str
+    description: str
+    def prepare_key(self, raw_data): ...
+    def generate_preset(self, enc_alg, key): ...
+    def wrap(self, enc_alg, headers, key, preset: Incomplete | None = None): ...
+    def unwrap(self, enc_alg, ek, headers, key): ...
+
+class RSAAlgorithm(JWEAlgorithm):
+    key_size: int
+    name: Incomplete
+    description: Incomplete
+    padding: Incomplete
+    def __init__(self, name, description, pad_fn) -> None: ...
+    def prepare_key(self, raw_data): ...
+    def generate_preset(self, enc_alg, key): ...
+    def wrap(self, enc_alg, headers, key, preset: Incomplete | None = None): ...
+    def unwrap(self, enc_alg, ek, headers, key): ...
+
+class AESAlgorithm(JWEAlgorithm):
+    name: Incomplete
+    description: Incomplete
+    key_size: Incomplete
+    def __init__(self, key_size) -> None: ...
+    def prepare_key(self, raw_data): ...
+    def generate_preset(self, enc_alg, key): ...
+    def wrap_cek(self, cek, key): ...
+    def wrap(self, enc_alg, headers, key, preset: Incomplete | None = None): ...
+    def unwrap(self, enc_alg, ek, headers, key): ...
+
+class AESGCMAlgorithm(JWEAlgorithm):
+    EXTRA_HEADERS: Incomplete
+    name: Incomplete
+    description: Incomplete
+    key_size: Incomplete
+    def __init__(self, key_size) -> None: ...
+    def prepare_key(self, raw_data): ...
+    def generate_preset(self, enc_alg, key): ...
+    def wrap(self, enc_alg, headers, key, preset: Incomplete | None = None): ...
+    def unwrap(self, enc_alg, ek, headers, key): ...
+
+class ECDHESAlgorithm(JWEAlgorithm):
+    EXTRA_HEADERS: Incomplete
+    ALLOWED_KEY_CLS = ECKey
```

Comment:
This one might need to be marked incomplete as well, it's assign to a tuple in `__init__.py`

https://github.com/lepture/authlib/blob/master/authlib/jose/__init__.py#L35